mitodl/mit-open
# Subscription management page

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4891


### Description (What does it do?)
Adds a "Preferences" page to the dashboard that contains a list of channels a user has subscribed to (from which they can unsubscribe)

### Screenshots (if appropriate):
<img width="1095" alt="Screenshot 2024-07-29 at 1 59 09 PM" src="https://github.com/user-attachments/assets/a4f52343-680d-4dea-bf27-3af8e306efce">
<img width="499" alt="Screenshot 2024-07-29 at 1 59 43 PM" src="https://github.com/user-attachments/assets/db20f5df-151e-462b-a005-d9087027453b">

### How can this be tested?
1. Checkout this branch
2. Subscribe to a few channels of different types (unit, department, topic)
3. go to the settings page by navigating to the dashboard and clicking "settings"
<img width="435" alt="Screenshot 2024-07-29 at 1 59 09 PM" src="https://github.com/user-attachments/assets/d67196e4-8ad8-4840-b56a-d2c790727565">

4. Unsubscribe from topics and see that it works as expected.

### Additional context
Functionally the unsubscribe buttons on this preferences page are the same as if you unsubscribed directly from the relevant channel's page. We are just consolidating all the subscriptions in one location here.
